src/libponyc/codegen/genopt.cc

```diff
@@ -179,11 +181,35 @@ class HeapToStack : public FunctionPass
     // TODO: for variable size alloca, don't insert at the beginning.
     Instruction* begin = &(*call.getCaller()->getEntryBlock().begin());
 
+#if PONY_LLVM < 500
     AllocaInst* replace = new AllocaInst(builder.getInt8Ty(), int_size, "",
       begin);
+#else
+    AllocaInst* replace = new AllocaInst(builder.getInt8Ty(),
+      (unsigned int)int_size->getZExtValue(), "", begin);
```

Comment:
The new signature of the array `AllocaInst` constructor is

```cpp
explicit AllocaInst(Type *Ty, unsigned AddrSpace,
                    Value *ArraySize = nullptr,
                    const Twine &Name = "",
                    Instruction *InsertBefore = nullptr);
```

This means that this is currently using the non-array constructor which isn't taking a size, and is passing the intended size as the address space. This call should instead be

```cpp
AllocaInst* replace = new AllocaInst(builder.getInt8Ty(), 0, int_size, "", begin);
```

Reply:
Oops, fixed now.